app/src/main/java/vn/mbm/phimp/me/Camera2.java

```diff
@@ -86,6 +100,58 @@ public Camera2() {
 
 	}
 
+	private boolean inRange(int SubjectValue, int High, int Low) {
+		// In case of a 360
+		SubjectValue =  (360 + (SubjectValue % 360)) % 360;
+		if (High < Low) {
+			return High <= SubjectValue && SubjectValue <= Low;
+		} else {
+			return High <= SubjectValue || SubjectValue <= Low;
+		}
+	}
+
+	private void setFlashIcon() {
+		switch (state) {
+			case 0:
+				flash.setImageResource(R.drawable.flash_on);
+				break;
+			case 1:
+				flash.setImageResource(R.drawable.flash_off);
+				break;
+			case 2:
+				flash.setImageResource(R.drawable.flash_auto);
+				break;
+			default:
+				flash.setImageResource(R.drawable.flash_on);
+				break;
+		}
+	}
+
+	private void rotateIcons(int degrees) {
+		flash.setRotation(degrees);
+		camera_switch.setRotation(degrees);
+		buttonClick.setRotation(degrees);
+	}
+
+	private void adjustIconPositions() {
+		if (inRange(camOrientation, NW, NE)) {
+			setFlashIcon();
+			rotateIcons(0);
+		} else if (inRange(camOrientation, NE, SE)) {
+			setFlashIcon();
+			rotateIcons(270);
+		} else if (inRange(camOrientation, SE, SW)) {
+			setFlashIcon();
+			rotateIcons(180);
+		} else if (inRange(camOrientation, SW, NW)) {
+			setFlashIcon();
+			rotateIcons(90);
+		} else {
+			setFlashIcon();
```

Comment:
As I suggested earlier, you are calling setFlashIcon on all if and else conditions, you should put this out of the condition stmt.